packages/compiler-core/src/transforms/hoistStatic.ts

```diff
@@ -168,18 +168,21 @@ export function getStaticType(
         }
 
         // check if any of the props contain runtime constants
-        if (returnType !== StaticType.HAS_RUNTIME_CONSTANT) {
-          for (let i = 0; i < node.props.length; i++) {
-            const p = node.props[i]
-            if (
-              p.type === NodeTypes.DIRECTIVE &&
-              p.name === 'bind' &&
-              p.exp &&
-              (p.exp.type === NodeTypes.COMPOUND_EXPRESSION ||
-                p.exp.isRuntimeConstant)
-            ) {
-              returnType = StaticType.HAS_RUNTIME_CONSTANT
-            }
+        for (
+          let i = 0;
+          returnType !== StaticType.HAS_RUNTIME_CONSTANT &&
```

Comment:
did you do a benchmark?
this code will check `returnType` every time.

Reply:
Obviously, my style would be quicker if the loop exited earlier, otherwise it would be slower because it will check returnType every time.
If only performance is considered, add a 'break' in the if branch of the original code is the fastest(add break to line 182).
But personally, I'd prefer my style because the code is more concise and the performance is not bad.
For benchmarks, I tested an array of 500,000 elements and exited when it looped to 250,000.
IE, old version of FF, old version of EDGE are the latter two styles fast. But chrome may be optimized internally in v8, which is not obvious.